src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -374,6 +374,26 @@ public char PeekChar(int delta)
             return ch;
         }
 
+        public char PreviousChar()
+        {
+            Debug.Assert(this.Position > 0);
+            var desiredPosition = this.Position - 1;
+            if (desiredPosition >= _basis && _offset > 1)
+            {
+                // _basis describes where in the source text the current chunk of characters starts at.
+                // So if the desired position is ahead of that, then we can just read the value out of
+                // the character window directly.
+                return this.CharacterWindow[_offset - 1];
```

Comment:
this is the common case.  The "previous character" is still within the chunk of characters from the source-text that is in the sliding window.